src/Raven.Client/Documents/Indexes/IndexDefinition.cs

```diff
@@ -330,11 +330,29 @@ public void RemoveDefaultValues()
                 Fields.Remove(key);
         }
 
+        private static readonly HashSet<(string Start, string End)> _commentTokens = new HashSet<(string Start, string End)>{("//",Environment.NewLine),("/*","*/")};
         public IndexType DetectStaticIndexType()
         {
             var firstMap = Maps.FirstOrDefault()?.TrimStart();
             if (firstMap == null)
                 throw new ArgumentNullException("Index definitions contains no Maps");
+            //Taking care for all comments combination before the first none comment line
+            int prevLength;
```

Comment:
This logic is fragile. 
I can have:
```
/*
foo 
*/
//  bar
map(
```

And the second one will not be detected. 
Better option is to just try to parse the string, as a C# expression. And if successful, check if the first expression is either a `from` query or a `docs` member reference.

Reply:
sure it will, why do you say it won't